### PR TITLE
Feat: [DS-189] Added status variants to RichIcon

### PR DESCRIPTION
**Pull request by vicky-comeau**

### CODE REVIEW

File: .changeset/sweet-cars-beam.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@hopper-ui/components": minor
+---
+
```
Comment (fraincs) [inline .changeset/sweet-cars-beam.md:4]: is this relevant?

File: .changeset/tricky-sloths-worry.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@hopper-ui/icons": minor
+---
+
```
Comment (fraincs) [inline .changeset/tricky-sloths-worry.md:4]: We should add the new options in the documentation as well

Comment (fraincs) [reply .changeset/tricky-sloths-worry.md:4]: we could also provide an example

File: .changeset/sweet-cars-beam.md (hunk 1 above)
Comment (vicky-comeau) [reply .changeset/sweet-cars-beam.md:4]: it's the missing changeset from my previous pr that I forgot

Comment (vicky-comeau) [reply .changeset/sweet-cars-beam.md:4]: I could put it in another pr, but I figured it would eb easier this way

Comment (fraincs) [reply .changeset/sweet-cars-beam.md:4]: that's alright

File: apps/docs/content/icons/react-icons/rich-icon-library.mdx [hunk 3]
```diff
@@ -13,7 +13,8 @@ All available rich icons in the icon library are shown below.
 
 Use the correct size for each icon. Icons should be used at their original size.
 
-The color of the rich icons can be changed by using the `variant` prop. The following [decorative colors](/tokens/semantic/color#decorative) are available as variants: `option1, option2, option3, option4, option5, option6, option7, option8`.
+The color of the rich icons can be changed by using the `variant` prop. The following [decorative colors](/tokens/semantic/color#decorative) are available as variants: `option1` to `option8`.
+The following status colors are available as well: `success`, `warning`, `danger`, `information` and `upsell`.
```
Comment (fraincs) [inline apps/docs/content/icons/react-icons/rich-icon-library.mdx:17]: We could link to status colors as well

File: apps/docs/content/icons/svg/rich-icon-library.mdx [hunk 4]
```diff
@@ -15,7 +15,7 @@ Use the correct size for each icon. Icons should be used at their original size.
 
 The color of the rich icons can be changed by using the right CSS Variables.
 
-The following variables are available for your styling needs: `--hop-RichIcon-placeholder-fill`, `--hop-RichIcon-placeholder-background` and `--hop-RichIcon-placeholder-shadow`. You should use the appropriate [decorative semantic tokens](/tokens/semantic/color#decorative), the available options for Rich Icons are `option1` to `option8`.
+The following variables are available for your styling needs: `--hop-RichIcon-placeholder-fill`, `--hop-RichIcon-placeholder-background` and `--hop-RichIcon-placeholder-shadow`. You should use the appropriate [decorative semantic tokens](/tokens/semantic/color#decorative) or status semantic tokens. The available options for Rich Icons are `option1` to `option8`, `success`, `warning`, `danger`, `information` and `upsell`.
```
Comment (fraincs) [inline apps/docs/content/icons/svg/rich-icon-library.mdx:18]: We could link to status colors as well.

File: apps/docs/content/icons/react-icons/rich-icon-library.mdx (hunk 3 above)
Comment (vicky-comeau) [reply apps/docs/content/icons/react-icons/rich-icon-library.mdx:17]: not really... they're not under the status category. It's a bit weird I know...

Comment (fraincs) [reply apps/docs/content/icons/react-icons/rich-icon-library.mdx:17]: oh alrighty :D

**Review by fraincs — APPROVED** [review]

**Pull request merged**